### ISSUE TITLE
Turn on lights initially for white balance

### ISSUE THREAD

**Issue by dmadison** [psieg/Lightpack#123]
At the moment, the lights are turned off in the setup wizard while the grab zones are re-arranged, but they are not re-enabled during the white balance adjustment (#95) unless one of the color sliders is moved.

It would be a nice quality of life change if the lights were enabled with the current calibration before the sliders are changed.

**Comment by psieg** [comment]
The thing is that there is no such thing as a global white balance. In reality, every grab zone has its own. Thus, when the wizard is started (unless all have the same value) there is no "old value"

**Comment by dmadison** [comment]
Right, but earlier in the wizard you select the profile to either create or modify. So the lights can be enabled with either the pre-existing values in the profile or using 100% on all channels. There's no reason the sliders need to move for the lights to come on.

**Comment by psieg** [comment]
So you're saying the lights do not turn on when the White Balance Page opens?

**Comment by dmadison** [comment]
Yes.

**Comment by psieg** [comment]
...with an Adalight device, right? That is weird, it works with my Lightpack(s)...
Not sure how to further investigate this...

**Comment by dmadison** [comment]
Yes, with an Adalight device. I hooked up the logic analyzer again, and it seems that the lights _are_ sent a few frames of color data when the wizard switches to the "White Balance" page. But all of the color information is '0'. Perhaps related to #124?

**Comment by psieg** [comment]
Afaik it always sends the maximum of led colors in the wizard. Are the first few ones white or are all of them black? Can you reproduce the problem with a brand new config (just rename all config files)?

**Comment by dmadison** [comment]
All of them are black. I just tested with a brand new config (renamed the "Prismatik" folder). A properly formatted frame of 255 LEDs is sent but all of the color data is '0' for all channels.

**Comment by psieg** [comment]
@dmadison does this still happen? With 5.11.2.14 it should send the correct number

**Comment by dmadison** [comment]
Just checked using a logic analyzer with 5.11.2.15: yes, this still happens. The correct number of LEDs is sent, but the initial frames are pure black.

**Comment by psieg** [comment]
I just debugged through the steps again. Yes, the first set of colors sent are in fact black (when settings are written they are applied to the current colors, which are black at the end of the configure device page), but the last frame does contain white colors, even without touching the sliders.

Can you confirm this?

**Comment by dmadison** [comment]
Confirmed. Testing with 5.11.2.17 and 10 LEDs, upon entering the calibration wizard Prismatik sends 6 frames of color data. The first 5 are black, and the sixth is white.

Unfortunately the frames are queued immediately after one-another, so the device drops every other frame (#117). Including the last one with the white color data :anguished:.

**Comment by psieg** [comment]
5216115138e0a9b4898ee4b2d281d00063ee793a should reduce that to two, but this probably won't even help if every other frame is dropped

**Comment by psieg** [comment]
@dmadison can you still reproduce this?

**Comment by roryslatter** [comment]
Hi, I'm having a similar issue. I'm using Adalight, 140 LED's and when I go through the installation wizard, all of the LED's turn off when the White Balance Wizard starts. I've tried moving the sliders but nothing happens. The lights turn on again when the White Balance Wizard closes.
I've uninstalled and reinstalled a few times to test this (clearing all config files each time) but I get the same result each time.
P.S. I love the changes/additions you've made to the original Prismatik!
